docs/_user_guide/best_practices_optimization.md

```diff
@@ -1,306 +1,354 @@
 You are welcome to redistribute it under certain conditions.
 Type 'license()' or 'licence()' for distribution details.
 
-  Natural language support but running in an English locale
-
 R is a collaborative project with many contributors.
 Type 'contributors()' for more information and
 'citation()' on how to cite R or R packages in publications.
 
 Type 'demo()' for some demos, 'help()' for on-line help, or
 'help.start()' for an HTML browser interface to help.
+
 Type 'q()' to quit R.
+>
+```
 
-> library('gurobi')
+```title="Terminal Command"
+library('gurobi')
+```
+```{ .yaml .no-copy title="Terminal Output" }
 Loading required package: slam
 >
 ```
 
 You can now run the R scripts to solve the optimization problem using Gurobi on interactive Yen nodes.
 
-**Important:** You need to load the `gurobi` and `R` modules every time you login to the Yens
-before running the interactive R scripts that use `gurobi` R package.
+!!! Warning
+    The default Gurobi 10 version is not compatible with the default R 4.3 version. To ensure proper functionality, you need to use R 4.2 instead. Running `ml gurobi R` with an incompatible version will result in an error. For more details on which Gurobi versions work with specific R versions, please refer to the information provided [here](https://support.gurobi.com/hc/en-us/articles/360025593712-Which-R-versions-are-supported-by-Gurobi). 
 
-Note, that if you want to use an older Gurobi version, you will need to install `gurobi` R package to your user
-R library. 
+!!! Important
+    You need to load the `gurobi` and `R` modules every time you log in to the Yens before running the interactive R scripts that use `gurobi` R package.
+
+Note that if you want to use an older Gurobi version, you will need to install `gurobi` R package to your user R library.
 
 Load R and the older Gurobi module:
 
+```title="Terminal Command"
+module purge
+ml R/4.2 gurobi/9.5.2
 ```
-$ module purge
-$ ml R gurobi/9.5.2
+
+Start R by typing `R`:
+
+```title="Terminal Command"
+R
 ```
 
-Start R by typing `R`, then install a personal `gurobi` package that links to Gurobi 9:
+Then install a personal `gurobi` package that links to Gurobi 9:
 
-```R
+```title="Terminal Command"
+install.packages("/software/non-free/Gurobi/gurobi952/linux64/R/gurobi_9.5-2_R_4.2.0.tar.gz", repos=NULL)
+```
+```{ .yaml .no-copy title="Terminal Output" }
 > install.packages("/software/non-free/Gurobi/gurobi952/linux64/R/gurobi_9.5-2_R_4.2.0.tar.gz", repos=NULL)
-Installing package into ‘/home/users/$USER/R/x86_64-pc-linux-gnu-library/4.2’
-(as ‘lib’ is unspecified)
+Warning in install.packages("/software/non-free/Gurobi/gurobi952/linux64/R/gurobi_9.5-2_R_4.2.0.tar.gz",  :
+  'lib = "/software/free/R/R-4.2.1/lib/R/library"' is not writable
+Would you like to use a personal library instead? (yes/No/cancel) yes
+Would you like to create a personal library
+‘/zfs/home/users/wolee/R/x86_64-pc-linux-gnu-library/4.2’
```

Comment:
You can use "Installing package into ‘/home/users/$USER/R/x86_64-pc-linux-gnu-library/4.2’" path in the warning that the user will see and add R as a language

Reply:
Could you clarify? Do you want me to add "Installing package into ‘/home/users/$USER/R/x86_64-pc-linux-gnu-library/4.2’" into the existing warning that was moved below the "Running Gurobi in R" title, or create a new warning below this output? Where should I add R as a language? To the output?

Reply:
Done! I also formatted a couple of Python outputs with Python language.